orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -342,6 +391,64 @@ void addObjectMethodsForTraining(py::module& m) {
         }
         ORT_THROW_IF_ERROR(static_cast<TrainingSession*>(sess->GetSessionHandle())->SetStateTensors(state_tensors, strict));
       })
+      .def("load_model_optimizer_state", [](PyTrainingSession* sess, std::unordered_map<std::string, std::unordered_map<std::string, py::object>>& model_state, std::unordered_map<std::string, std::unordered_map<std::string, py::object>>& opt_state, bool strict) {
+        std::unordered_map<std::string, NameMLValMap> model_state_tensors;
+        for (const auto& fp_or_mp_tensors: model_state) {
+          NameMLValMap model_tensors;
+          for (auto initializer : fp_or_mp_tensors.second) {
+            OrtValue ml_value;
+            auto px = sess->GetSessionHandle()->GetModelInputs();
+            if (!px.first.IsOK() || !px.second) {
+              throw std::runtime_error("Either failed to get model inputs from the session object or the input def list was null");
+            }
+            CreateGenericMLValue(px.second, GetAllocator(), initializer.first, initializer.second, &ml_value);
+            if (PyErr_Occurred()) {
+              PyObject *ptype, *pvalue, *ptraceback;
+              PyErr_Fetch(&ptype, &pvalue, &ptraceback);
+
+              PyObject* pStr = PyObject_Str(ptype);
+              std::string sType = py::reinterpret_borrow<py::str>(pStr);
+              Py_XDECREF(pStr);
+              pStr = PyObject_Str(pvalue);
+              sType += ": ";
+              sType += py::reinterpret_borrow<py::str>(pStr);
+              Py_XDECREF(pStr);
+              throw std::runtime_error(sType);
+            }
+            model_tensors.insert(std::make_pair(initializer.first, ml_value));
+          }
+          model_state_tensors.insert(std::make_pair(fp_or_mp_tensors.first, model_tensors));
+        }
+        std::unordered_map<std::string, NameMLValMap> opt_tensors;
+        for (auto weight: opt_state) {
+            std::string weight_name = weight.first;
+            NameMLValMap opt_per_weight;
+            for (auto initializer : weight.second) {
+            OrtValue ml_value;
+            auto px = sess->GetSessionHandle()->GetModelInputs();
+            if (!px.first.IsOK() || !px.second) {
+              throw std::runtime_error("Either failed to get model inputs from the session object or the input def list was null");
+            }
+            CreateGenericMLValue(px.second, GetAllocator(), initializer.first, initializer.second, &ml_value);
+            if (PyErr_Occurred()) {
```

Comment:
This block can be replaced by method `ThrowIfPyErrOccured`

Reply:
Good point. Replaced block with `ThrowIfPyErrOccured`